### PR TITLE
Make Review with tests

### PR DESCRIPTION
**Pull request by maciasj**
# Description

## Context

(Optional)

Current situation of the system, explanation why this PR is needed.

## Goal

Explanation about what we want to achieve with this PR.

## Support documents

- Taiga Ticket: [ticket](link)

- UI/UX design: [link](https://paack.co)

## Design considerations

(Optional)

Section dedicated to explain what design considerations we had while coding this PR.

# Testing

Section dedicated to explain how and what to test in this PR.

# Configuration

## New environment

(Optional)

Environment variables added/changed that needs to be added manually by the other developers

## New dependencies

(Optional)

New packages/libraries added / updated

# Attachments

(Optional)

- <details closed>
    
  </details>

Section dedicated to add any screenshots or other documents we find relevant to help understand or analyze this PR.

# Dependencies or blockers

(Optional: List of dependencies or blockers that make this PR not yet ready to be merged with the main branch)
## Subtasks

(List of subtask PRs needed for this feature)

- [ ] #XX
...

### CODE REVIEW

File: src/domain/repositories/event-repository/event-repository.ts [hunk 1]
```diff
@@ -36,13 +38,22 @@ export class EventRepository {
...
 
-
+  public static async modifyValoracions(
+    event: IEvent,
+    newValoracions: IReview[],
+  ): Promise<void> {
+    
+     await EventModel.findByIdAndUpdate( event.id ,
+     { valoracions: newValoracions }, {new:true});
+  }
```
Comment (quimvi) [inline src/domain/repositories/event-repository/event-repository.ts:56]: Esto no hace falta, ya tienes el editar Event. El repositorio piensa que es una clase que es como una base de datos, crea, modifica deletea... No podemos tener 1 funcion de guardar por cada parametro de event. Tenemos 1 funcion de edit event y ya esta

File: src/application/use-cases/makeEventReview/makeReview.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          if (valo.author === newIReview.author) {
+            res.status(404).json({
+              message: `Usuario ${valo.author} ya ha valorado este evento`,
+            });
+            return;
+            }
+          }
+      newValoracions.push(newIReview);
+      try{
+      await EventRepository.modifyValoracions(event,[...oldValoracions, ...newValoracions] );
```
Comment (quimvi) [inline src/application/use-cases/makeEventReview/makeReview.ts:31]: Aqui solo haria falta entonces que hagas el editar event del EventRepository

File: src/domain/entities/event/event.ts [hunk 3]
```diff
@@ -54,4 +58,7 @@ export class Event implements IEvent {
     const ids = this.participants.map((participant) => participant.username);
     return ids;
   }
-}
+  public updateValoracions(newValoracions: IReview[]): void{
+    this.valoracions = newValoracions;
+  } 
+}
```
Comment (quimvi) [inline src/domain/entities/event/event.ts:64]: Esto realmente lo puedes cambiar para que haga un push de valoracions, de esta forma no hara falta que gestiones el array en el use case, vas haciendp `EVENT.afegirValoracio(newValoracio)`, y luego `eventRepositoryu.editarEvent(EVENT)`

File: src/domain/repositories/event-repository/event-repository.ts (hunk 1 above)
Comment (maciasj) [reply src/domain/repositories/event-repository/event-repository.ts:56]: okayy

**Review by quimvinals — APPROVED** [review]

**Comment by quimvinals** [comment]
de puta madre

**Pull request merged**